### PR TITLE
Goto links for LSP hover

### PR DESCRIPTION
**Pull request by sdankel**
## Description

This is a subset of https://github.com/FuelLabs/sway/pull/4532/files. I'm trying to narrow down why some of the e2e tests are failing. This PR contains the changes for the "Go to type" links only, excluding the changes in sway-core that were needed for "Goto implementations" to work.

Closes https://github.com/FuelLabs/sway/issues/2852

image

## Checklist

- [x] I have linked to any relevant issues.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation where relevant (API docs, the reference, and the Sway book).
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added (or requested a maintainer to add) the necessary `Breaking*` or `New Feature` labels where relevant.
- [x] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [ ] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

File: sway-lsp/src/core/session.rs [hunk 1]
```diff
@@ -26,7 +26,7 @@ use sway_core::{
     language::{
         lexed::LexedProgram,
         parsed::{AstNode, ParseProgram},
-        ty,
+        ty::{self},
```
Comment (JoshuaBatty) [inline sway-lsp/src/core/session.rs:29]: same

**Review by JoshuaBatty — APPROVED** [review]
Nice work, look good to me. Just one tiny nit but it doesn't need to hold up this PR.

**Review by eureka-cpu — APPROVED** [review]
LGTM 🚢

File: sway-lsp/src/capabilities/hover/mod.rs [hunk 2]
```diff
@@ -9,12 +11,18 @@ use crate::{
 };
 use std::sync::Arc;
 use sway_core::{
-    language::{ty, Visibility},
+    language::{
+        ty::{self},
```
[the remark this answers is not in the document]
Comment (sdankel) [reply sway-lsp/src/capabilities/hover/mod.rs:15]: Thanks, will address in the next PR!

**Pull request merged**